src/main/java/seedu/lifeasier/commands/EditNotesCommand.java

```diff
@@ -0,0 +1,145 @@
+import seedu.lifeasier.notes.NoteCommandFunctions;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+public class EditNotesCommand extends Command {
+    private static Logger logger = Logger.getLogger(EditNotesCommand.class.getName());
+    private String title;
+
+    public EditNotesCommand(String title) {
+        this.title = title;
+    }
+
+    private void findTitle(Ui ui, NoteList notes, String title) throws TitleNotFoundException {
+        logger.log(Level.INFO, "Start for finding title in note list");
+        int noteNumber = -1;
+        int matchNumber = 0;
+
+        for (int i = 0; i < notes.size(); i++) {
+            if (notes.get(i).getTitle().contains(title)) {
+                matchNumber++;
+                noteNumber = i;
+            }
+        }
+        logger.log(Level.INFO, "End for finding title in note list");
+
+        switch (matchNumber) {
+        case 0:     // no matches
+            logger.log(Level.SEVERE, "No matches for title");
+            throw new TitleNotFoundException();
+        case 1:
+            logger.log(Level.INFO, "One match found");
+            System.out.println(notes.get(noteNumber).toString());
+            ui.showConfirmEditMessage();
+            parseUserResponse(ui, notes, noteNumber, ui.readCommand());
+            break;
+        default:
+            logger.log(Level.INFO, "Multiple matches found");
+            ui.showMultipleMatchesFoundMessage();
+
+            logger.log(Level.INFO, "Start of printing all matching notes");
+            NoteCommandFunctions.printMultipleMatches(notes, title);
+            logger.log(Level.INFO, "End of printing all matching notes");
+
+            noteNumber = Integer.parseInt(ui.readCommand());
+            NoteCommandFunctions.checkForIndexOutOfBounds(notes, noteNumber);
+
+            System.out.println(notes.get(noteNumber - 1).toString());
+            ui.showConfirmEditMessage();
+            parseUserResponse(ui, notes, noteNumber - 1, ui.readCommand());
+        }
+
+    }
+
+    private void parseUserResponse(Ui ui, NoteList notes, int noteNumber, String input) {
+        logger.log(Level.INFO, "Start check for Y/N input");
+        while (!input.trim().equals("Y") && !input.trim().equals("N")) {
+            ui.showInvalidConfirmationMessage();
+            input = ui.readCommand();
+        }
+        logger.log(Level.INFO, "End check for Y/N input");
+
+        if (input.trim().equals("Y")) {
+            logger.log(Level.INFO, "Y is inputted");
+            ui.showEditWhichPartMessage();
+            while (!input.trim().equals("T") && !input.trim().equals("D")) {
+                ui.showInvalidTitleDescriptionConfirmationMessage();
+                input = ui.readCommand();
+            }
+            changeTitleOrDescription(ui, notes, noteNumber, input);
+        } else {
+            logger.log(Level.INFO, "N is inputted");
+            ui.showNoteNotEditedMessage();
+        }
+
+    }
+
```

Comment:
You could consider refactoring this into the Parser class, and passing the parser object into the execute method too.
Doing this could help in ensuring SRP.